cluster-autoscaler/core/scaledown/actuation/group_deletion_scheduler.go

```diff
@@ -89,8 +89,18 @@ func (ds *GroupDeletionScheduler) ScheduleDeletion(nodeInfo *framework.NodeInfo,
 
 	nodeDeleteResult := ds.prepareNodeForDeletion(nodeInfo, drain)
 	if nodeDeleteResult.Err != nil {
-		ds.AbortNodeDeletion(nodeInfo.Node(), nodeGroup.Id(), drain, "prepareNodeForDeletion failed", nodeDeleteResult)
-		return
+		isTimeout := true
+		for _, podEvictionResult := range nodeDeleteResult.PodEvictionResults {
```

Comment:
IIUC we should use this logic only if `nodeDeleteResult.ResultType == status.NodeDeleteErrorFailedToEvictPods`. In other cases the issue was not caused by timed out drain of pods, and so we should abort deletion.